module/communication/mqtt/src/main/java/org/openbase/jul/communication/mqtt/RPCServerImpl.kt

```diff
@@ -142,8 +142,8 @@ class RPCServerImpl(scope: Scope, config: CommunicatorConfig) : RPCCommunicatorI
             responseBuilder.result = result
         } catch (ex: Exception) {
             when (ex) {
-                is InvocationTargetException, is CouldNotPerformException -> responseBuilder.error =
-                    ex.stackTraceToString()
+                is InvocationTargetException -> responseBuilder.error =
+                    ex.cause!!.stackTraceToString()
```

Comment:
Try to avoid usage of `!!` in kotlin since the cause can be null, right? Thus, it would lead to a `Nullpointer` again. Just use something like `ex?.cause?.stackTraceToString()?:"UNKNOWN_ERROR"` or just `ex?.cause?.stackTraceToString()` in case `ressponseBuilder.errer` is nullable.

Reply:
:+1: